Gems/ScriptCanvas/Code/Include/ScriptCanvas/Libraries/String/Deprecated/StringGenerics.h

```diff
@@ -21,14 +21,16 @@ namespace ScriptCanvas
             AZStd::to_lower(sourceString.begin(), sourceString.end());
             return sourceString;
         }
-        SCRIPT_CANVAS_GENERIC_FUNCTION_NODE(ToLower, k_categoryName, "{FC5FA07E-C65D-470A-BEFA-714EF8103866}", "Makes all the characters in the string lower case", "Source");
+        SCRIPT_CANVAS_GENERIC_FUNCTION_NODE_TRAIT(ToLower, ScriptCanvas::NoDefaultArguments, k_categoryName, "{FC5FA07E-C65D-470A-BEFA-714EF8103866}", true, "Makes all the characters in the string lower case", "Source");
+        SCRIPT_CANVAS_GENERIC_FUNCTION_REPLACEMENT_CONFIG(ToLower, ScriptCanvas::NoDefaultArguments, "ScriptCanvas_StringFunctions_ToLower", "Source", "String");
 
         AZ_INLINE Data::StringType ToUpper(Data::StringType sourceString)
         {
             AZStd::to_upper(sourceString.begin(), sourceString.end());
             return sourceString;
         }
-        SCRIPT_CANVAS_GENERIC_FUNCTION_NODE(ToUpper, k_categoryName, "{323951D4-9BB1-47C9-BD2C-2DD2B750217F}", "Makes all the characters in the string upper case", "Source");
+        SCRIPT_CANVAS_GENERIC_FUNCTION_NODE_TRAIT(ToUpper, ScriptCanvas::NoDefaultArguments, k_categoryName, "{323951D4-9BB1-47C9-BD2C-2DD2B750217F}", true, "Makes all the characters in the string upper case", "Source");
```

Comment:
We definitely need to combine these MACROS, or eliminate one. There should be no use of the traits class in the replacement macro (otherwise, it still bloats the compile times and sizes).  SCRIPT_CANVAS_GENERIC_FUNCTION_REPLACEMENT_CONFIG should the old function name, "ToLower", the old UUID, and that should be it. It should create an empty node that derives from Node, only to reflect and supply the new UUID. The registrar code may also need a replacement, to call reflect on all those nodes. They only get reflected to the serialize context. When any graph is loaded in the editor or updater, the old slot names and types will be trusted to match against the replacements specified in the code gen. Then you can delete even all of the old function definitions themselves, instead of having to leave them to drive the template instantiation.
```suggestion
        SCRIPT_CANVAS_GENERIC_FUNCTION_NODE_TRAIT(ToUpper, ScriptCanvas::NoDefaultArguments, k_categoryName, "{323951D4-9BB1-47C9-BD2C-2DD2B750217F}", true, "Makes all the characters in the string upper case", "Source");
```

Reply:
save this advice for your next PR